server/controllers/DatabaseController.js

```diff
@@ -0,0 +1,242 @@
+    exec(pgDumpCommand, (error, stdout, stderr) => {
+      if (error) {
+        return res
+          .status(400)
+          .send(`Error executing pg_dump: ${sanitizeError(error.message)}`);
+      }
+      if (stderr) {
+        return res
+          .status(400)
+          .send(`pg_dump stderr:: ${sanitizeError(stderr)}`);
+      }
+      return res
+        .status(200)
+        .send(`Database dump completed successfully: ${DUMP_FILE}`);
+    });
+  } catch (error) {
+    return res
+      .status(400)
+      .send(`Unable to dump database: ${sanitizeError(error.message)}`);
+  }
+};
+
+const restorePostgresDatabase = async (req, res) => {
+  // Prevent unintentionally dropping or restoring database tables if ran on
+  // production unless manually done
+  if (process.env.ENVIRONMENT === 'production') {
+    return res.status(405).send('This request is not permitted');
+  }
+
+  const { pathToFile } = req.body;
+  if (!pathToFile)
+    return res.status(400).send("'pathToFile' is missing. Please provide.");
+
+  // Purge the database's tables and functions to make restoring with the
+  // pg import easier
+  await removeDatabaseTablesAndFunctions(res);
+
+  try {
+    // delete the tables currently in the db
+    const pgImportCommand = `PGPASSWORD=${DB_PASSWORD} psql -U ${DB_USER} -h ${DB_HOST} -d ${DB_NAME} -f ${pathToFile}`;
+
+    // Execute the command
+    exec(pgImportCommand, (error, stdout, stderr) => {
+      if (error) {
+        res
+          .status(400)
+          .write(`Error executing pg import: ${sanitizeError(error.message)}`);
+      }
+      if (stderr) {
+        res.status(400).write(`pg import stderr:: ${sanitizeError(stderr)}`);
+      }
+      res.status(200).write(`Database import completed successfully`);
+      return res.end();
+    });
+  } catch (error) {
+    res
+      .status(400)
+      .write(`Unable to import database: ${sanitizeError(error.message)}`);
+    return res.end();
+  }
+};
+
+// Function to clean up the dump folder
+const cleanFolder = (req, res) => {
+  const { dumpOutputDirectory, maxFiles } = req.body;
+
+  if (maxFiles) {
+    if (!isNaN(maxFiles) && Number.isInteger(Number(maxFiles))) {
+      // continue
+    } else {
+      return res.status(400).send('Unable to parse maxFiles value provided');
+    }
+  } else {
+    // continue
+  }
+
+  // Default the output directory to ~/db_dumps if no path provided
+  const OUTPUT_DIR = dumpOutputDirectory || path.join(os.homedir(), 'db_dumps');
+  // Default the number of max files to 28 if no value provided
+  const MAX_FILES = Number(maxFiles) || 28;
+
+  try {
+    // Read all files in the folder
+    const files = fs.readdirSync(OUTPUT_DIR).map(file => {
+      const filePath = path.join(OUTPUT_DIR, file);
+      return {
+        name: file,
+        path: filePath,
+        time: fs.statSync(filePath).mtime.getTime() // Get last modified time
+      };
+    });
+
+    // Sort files by modification time (oldest first)
```

Comment:
Some of these comments feel unnecessary when the code is self-documenting or straightforward.